README.md
# clean_swiper

Clean Swiper is a starting point for clean and simple swiper. 
Whenever you just need to swipe multiply elements and you don't need any
fancy features clean_swiper is here for you.

![Clean Swiper Exmaple](https://raw.githubusercontent.com/clean/clean_swiper/master/doc/showcase.gif)

## Installing

Include `clean_swiper` in your `pubspec.yaml` file:

```yaml
dependencies:
  flutter:
    sdk: flutter
  clean_swiper: <latest_version>
```

If your IDE doesn't do it automatically, type:

`flutter packages get`


## Using

```dart
import 'package:clean_swiper/clean_swiper.dart';

Container(
  height: 250,
  child: CleanSwiper(
    viewportFraction: 0.9,
    initialPage: 0,
    children: <Widget>[
      Card(
        color: Colors.yellow,
        child: Center(child: Text('Card 1'))
      ),
      Card(
        color: Colors.pink,
        child: Center(child: Text('Card 2'))
      ),
      Card(
        color: Colors.teal,
        child: Center(child: Text('Card 3'))
      ),
    ]
  ),
)
...
```

CHANGELOG.md
## [0.0.5] - 19 Nov 2019

* Allow to pass own controller to CleanSwiper

## [0.0.4] - 19 Nov 2019

* Code formatting based on pub.dev qa checks

## [0.0.3] - 19 Nov 2019

* Added example usage
* Added readme

## [0.0.2] - 17 Nov 2019

* initalPage param added

## [0.0.1] - 17 Nov 2019

* Created `clean_swiper` which provides basic swipe widget feature


test/clean_swiper_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:clean_swiper/clean_swiper.dart';

void main() {
  testWidgets('CleanSwiper instance check', (WidgetTester tester) async {
    // Create the widget by telling the tester to build it.
    await tester.pumpWidget(
      MaterialApp(
        home: CleanSwiper(
          children: <Widget>[
            Container(),
            Container(),
          ],
        ))
      );
    expect(find.byType(CleanSwiper), findsOneWidget);
  });
}


lib/clean_swiper.dart
library clean_swiper;

import 'dart:ui';

import 'package:flutter/widgets.dart';

class CleanSwiper extends StatefulWidget {
  @override
  _CleanSwiperState createState() => _CleanSwiperState();

  final List<Widget> children;
  final double viewportFraction;
  final int initialPage;
  PageController controller;

  CleanSwiper(
      {Key key,
      @required this.children,
      this.controller,
      this.viewportFraction = 0.8,
      this.initialPage = 0})
      : super(key: key);
}

class _CleanSwiperState extends State<CleanSwiper> {
  PageController controller;

  @override
  void initState() {
    super.initState();

    (widget.controller != null)
      ? controller = widget.controller
      : controller = PageController(
          initialPage: widget.initialPage,
          viewportFraction: widget.viewportFraction);
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return PageView.builder(
      controller: controller,
      itemCount: widget.children.length,
      itemBuilder: (context, position) {
        return widget.children[position];
      },
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:clean_swiper/clean_swiper.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Clean Swiper Demo',
      home: ExamplePage()
    );
  }
}

class ExamplePage extends StatelessWidget {
  final _pageController = PageController(
    initialPage: 1
  );

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Clean Swiper Example')
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          Container(
              height: 200,
              child: CleanSwiper(
                viewportFraction: 0.9,
                children: <Widget>[
                  Container(
                    margin: EdgeInsets.symmetric(horizontal: 5),
                    decoration: BoxDecoration(
                      image: DecorationImage(
                        image: NetworkImage("https://images.pexels.com/photos/257840/pexels-photo-257840.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"),
                        fit: BoxFit.cover,
                      )
                    ),
                  ),
                  Container(
                    margin: EdgeInsets.symmetric(horizontal: 5),
                    decoration: BoxDecoration(
                      image: DecorationImage(
                        image: NetworkImage("https://images.pexels.com/photos/33109/fall-autumn-red-season.jpg?auto=compress&cs=tinysrgb&dpr=2&w=500"),
                        fit: BoxFit.cover,
                      )
                    ),
                  ),
                  Container(
                    margin: EdgeInsets.symmetric(horizontal: 5),
                    decoration: BoxDecoration(
                      image: DecorationImage(
                        image: NetworkImage("https://images.pexels.com/photos/814499/pexels-photo-814499.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"),
                        fit: BoxFit.cover,
                      )
                    ),
                  ),
                ],
              ),
          ),
          Container(
              height: 250,
              child: CleanSwiper(
                viewportFraction: 0.9,
                children: <Widget>[
                  Card(
                    color: Colors.yellow,
                    child: Center(child: Text('Card 1'))
                  ),
                  Card(
                    color: Colors.pink,
                    child: Center(child: Text('Card 2'))
                  ),
                  Card(
                    color: Colors.teal,
                    child: Center(child: Text('Card 3'))
                  ),
                ],
              ),
          ),
          Container(
            margin: EdgeInsets.symmetric(vertical: 50),
            height: 30,
            color: Colors.white38,
            child: CleanSwiper(
              controller: _pageController,
              viewportFraction: 1,
              children: <Widget>[
                Text('Text 1', textAlign: TextAlign.center,),
                Text('Text 2', textAlign: TextAlign.center,),
                Text('Text 3', textAlign: TextAlign.center,),
              ],
            )
          )
        ],
      )
      
    );
  }
}